torch/csrc/jit/script/compiler.cpp

```diff
@@ -578,8 +578,9 @@ c10::optional<MatchedSchema> tryMatchSchema(
 
   // if we finish the loop will we have consumed all arguments?
   size_t used_args = 0;
-  for (size_t schema_i = 0; schema_i < schema.arguments().size(); ++schema_i) {
-    const auto& arg = schema.arguments()[schema_i];
+
+  auto processArgument = [&](const Argument& arg, size_t schema_i) -> Value* {
```

Comment:
I don't think wrapping this logic in a lambda is an improvement. It makes the control flow wonky, mutably captures some references, and "processArgument" isn't a descriptive name. This (plus the long list of params to this function) seems like a sign that we should restructure schema matching.